src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,316 @@
+     * @throws DukeException Error message to indicate invalid deadline command
+     */
+    private static String generateDeadlineResponse(String[] tokens, TaskList taskList) throws DukeException {
+        if (tokens.length < 2) {
+            throw new DukeException("The description of deadline cannot be empty.");
+        }
+
+        String[] parts = tokens[1].split("/by", 2);
+        if (parts.length < 2 || parts[0].trim().isEmpty() || parts[1].trim().isEmpty()) {
+            throw new DukeException("Please provide a task description and deadline.");
+        }
+
+        String taskDescription = parts[0].trim();
+        String dueByString = parts[1].trim();
+
+        try {
+            LocalDate dueBy = LocalDate.parse(dueByString, DateTimeFormatter.ofPattern("dd/MM/yyyy"));
+
+            Task newTask = new Deadline(taskDescription, dueBy, "0");
+            taskList.addTask(newTask);
+
+            return generateTaskAddedResponse(newTask, taskList);
+        } catch (DateTimeException e) {
+            throw new DukeException("Please enter valid date in dd/MM/yyyy format.");
+        }
+
+    }
+
+    /**
+     * Generates the response for an event command
+     * @param tokens The string array containing the command split into parts
+     * @param taskList Given TaskList of the user
+     * @return String containing the event added response
+     * @throws DukeException Error message to indicate invalid event command
+     */
+    private static String generateEventResponse(String[] tokens, TaskList taskList) throws DukeException {
+        if (tokens.length < 2) {
+            throw new DukeException("The description of event cannot be empty.");
+        }
+
+        String[] parts = tokens[1].split("/from", 2);
+        if (parts.length < 2 || parts[0].trim().isEmpty() || parts[1].trim().isEmpty()) {
+            throw new DukeException("Please provide a task description and event timing.");
+        }
+
+        String[] timingParts = parts[1].split("/to", 2);
+        if (timingParts.length < 2 || timingParts[0].trim().isEmpty() || timingParts[1].trim().isEmpty()) {
+            throw new DukeException("Please provide both the start and end timings for the event.");
+        }
+
+        String taskDescription = parts[0].trim();
+        String startTimingString = timingParts[0].trim();
+        String endTimingString = timingParts[1].trim();
+
+        try {
+            LocalDate startTiming = LocalDate.parse(startTimingString, DateTimeFormatter.ofPattern("dd/MM/yyyy"));
+            LocalDate endTiming = LocalDate.parse(endTimingString, DateTimeFormatter.ofPattern("dd/MM/yyyy"));
+
+            Task newTask = new Event(taskDescription, startTiming, endTiming, "0");
+            taskList.addTask(newTask);
+
+            return generateTaskAddedResponse(newTask, taskList);
+
+        } catch (DateTimeException e) {
+            throw new DukeException("Please enter valid date in dd/MM/yyyy format.");
+        }
+
+    }
```

Comment:
This method looks a bit too long (32 LoC). Maybe consider refactoring it to extract a more general command parser? Because I noticed some repetition in your code in this file, so maybe extracting the repetitive part out is a good idea!